llama/llama.go

```diff
@@ -680,3 +681,32 @@ func (s *SamplingContext) Sample(llamaContext *Context, idx int) int {
 func (s *SamplingContext) Accept(id int, applyGrammar bool) {
 	C.gpt_sampler_caccept(s.c, C.llama_token(id), C.bool(applyGrammar))
 }
+
+type JsonSchema struct {
```

Comment:
Add a test or two?

Reply:
Unfortunately cannot add much here since it's just calling the c function for the most part in `AsGrammar`

Reply:
A few worthwhile cases to test:

1. Invalid should return ""
2. Valid case